### ISSUE TITLE
Problem loading pretrained model

### ISSUE THREAD

**Issue by guillaume-heusch** [delyan-boychev/imaginet#2]
Hi ! 

First of all, thanks for your work and this repo !

Out of curiosity, I tried to run your model on one single image using your pretrained checkpoint, but I was not able to do so. 

Here's my code (basically, it's a copy from `./testing_utils/testing/ImagiNet/imaginet_model.py`)

```python
ckpt = torch.load("./required_libs/imaginet_weights.pt", map_location="cpu")

model = ConResNet(name="resnet50nodown", selfcon_pos=[False, True, False], selfcon_size="fc", dataset="imaginet")
model.load_state_dict(ckpt["model"])

classifier = nn.Sequential(nn.BatchNorm1d(2048), nn.Linear(2048, 1024), nn.ReLU(), nn.Dropout(0.2), nn.Linear(1024, 1))
classifier.load_state_dict(ckpt["classifier"])
```

And here's the error I'm getting:

```bash
Error running imaginet: running_mean should contain 128 elements not 2048
```
Any idea why is it so ?

Also, another quick question: what image size does your model expect ?

Thanks and best regards !

**Comment by delyan-boychev** [comment]
Hello @guillaume-heusch,
First, thanks for your interest in our work!

I tried to reproduce your problem, but when loading with the following snippet using the `ConResNet` from `training_utils/networks`, I didn't encounter any issues. This is the code snippet, I used after running `testing_utils/testing/setup_testing.py`:
```python
import os
import sys

repo_path = os.path.abspath("../../training_utils") 
sys.path.insert(0, repo_path)

import torch
import torch.nn as nn
import torchvision.models as models
from torchvision import transforms
from components.datasets import ImagiNet_Testset
import numpy as np
from networks.resnet_big import ConResNet
from torch.utils.data import DataLoader
from tqdm import tqdm
import random
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import roc_auc_score, balanced_accuracy_score
from scipy.special import expit
import csv

random.seed(42)
torch.manual_seed(42)
torch.cuda.manual_seed(42)
np.random.seed(42)

mean = (0.485, 0.456, 0.406)
std = (0.229, 0.224, 0.225)
norm = transforms.Normalize(mean, std)
transform =  transforms.Compose([transforms.ToTensor(), norm])


model = ConResNet(name="resnet50nodown", selfcon_pos=[False, True, False], selfcon_size="fc", dataset="imaginet")
state = torch.load("../required_libs/imaginet_weights.pt", map_location="cpu")
model.load_state_dict(state["model"])
model = model.eval()
classifier = nn.Sequential(nn.BatchNorm1d(2048), nn.Linear(2048, 1024), nn.ReLU(), nn.Dropout(0.2), nn.Linear(1024, 1))
classifier.load_state_dict(state["classifier"])
classifier.eval()
print("Model loaded correctly!")
```

This is the result of the script run:
```console
delyan-boychev@Delyans-MacBook-Air ~/D/i/t/testing (main)> python3 ImagiNet/imaginet_model.py                                                                                                        (sink) 
Model loaded correctly!
delyan-boychev@Delyans-MacBook-Air ~/D/i/t/testing (main)>   
```

Make sure to use the right checkpoint located [here](https://drive.google.com/file/d/1kI-j2J8JinyRBdemfPyPMhNRE79FlsgX/view). The resolution we use is $`256 \times 256`$ after crop (no resizing for better accuracy). You can try multiple random crops if the original resolution is high, then average the scores across them. If you have any other questions, I am open to answering them. If the problem reappears, reopen the issue with more details!

**Comment by guillaume-heusch** [comment]
Hi @delyan-boychev 

Thanks for your help ! 
My error is fixed now (and it was when _using_ the model, and not _loading_ the model actually ).
